build/commands/lib/util.js

```diff
@@ -606,48 +607,90 @@ const util = {
     }
 
     Log.progressStart('build redirect_cc')
-    gnArgs = {
+    const buildArgs = {
       'import("//brave/tools/redirect_cc/args.gni")': null,
       use_remoteexec: config.useRemoteExec,
       rbe_exec_root: config.rbeExecRoot,
       rbe_bin_dir: config.realRewrapperDir,
       real_rewrapper: path.join(config.realRewrapperDir, 'rewrapper'),
     }
 
-    const buildArgsStr = util.buildArgsToString(gnArgs)
-    util.run('gn', ['gen', config.nativeRedirectCCDir, '--args="' + buildArgsStr + '"'], options)
-
+    util.runGnGen(config.nativeRedirectCCDir, buildArgs)
     util.buildTarget('brave/tools/redirect_cc', mergeWithDefault({outputDir: config.nativeRedirectCCDir}))
     Log.progressFinish('build redirect_cc')
   },
 
-  runGnGen: (options) => {
-    const buildArgsStr = util.buildArgsToString(config.buildArgs())
-    const buildArgsFile = path.join(config.outputDir, 'brave_build_args.txt')
-    const buildNinjaFile = path.join(config.outputDir, 'build.ninja')
-    const gnArgsFile = path.join(config.outputDir, 'args.gn')
-    const prevBuildArgs = fs.existsSync(buildArgsFile) ?
-      fs.readFileSync(buildArgsFile) : undefined
-    const extraGnGenOptsFile = path.join(config.outputDir, 'brave_extra_gn_gen_opts.txt')
-    const prevExtraGnGenOpts = fs.existsSync(extraGnGenOptsFile) ?
-      fs.readFileSync(extraGnGenOptsFile) : undefined
+  runGnGen: (outputDir, buildArgs, extraGnGenOpts = [], options = config.defaultOptions) => {
+    // Store extraGnGenOpts in buildArgs as a comment to rerun gn gen on change.
+    assert(Array.isArray(extraGnGenOpts))
+    if (extraGnGenOpts.length) {
+      buildArgs['# extraGnGenOpts'] = extraGnGenOpts
```

Comment:
Could we add `.join(',')` to avoid saving a list in `buildArgs`?
With `JSON.stringify()` it works well, but in terms of typing using list make the code sophisticated.

Reply:
Rewrote this generation a little to look like this:
```
...
enable_brave_page_graph=true
enable_brave_page_graph_webapi_probes=true
# Extra gn gen options: --ide=vs2022
```